[BUG]: Cannot install Giotto-tda: Failed to build giotto-tda ERROR: Could not build wheels for giotto-tda, which is required to install pyproject.toml-based projects

**Describe the bug**
<!--
A clear and concise description of what the bug is.
Example: "Joblib error thrown when calling fit on VietorisRipsPersistence."
-->
I tried to install giotto-tda with `pip install giotto-tda`, but had the following error:
```
Collecting giotto-tda
  Using cached giotto-tda-0.1.4.tar.gz (99 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy>=1.17.0 in /Users/naf/Library/Python/3.11/lib/python/site-packages (from giotto-tda) (1.26.1)
Requirement already satisfied: scipy>=0.17.0 in /opt/homebrew/lib/python3.11/site-packages (from giotto-tda) (1.11.3)
Requirement already satisfied: joblib>=0.11 in /opt/homebrew/lib/python3.11/site-packages (from giotto-tda) (1.3.2)
Requirement already satisfied: scikit-learn>=0.22.0 in /opt/homebrew/lib/python3.11/site-packages (from giotto-tda) (1.3.1)
Requirement already satisfied: matplotlib>=3.0.3 in /opt/homebrew/lib/python3.11/site-packages (from giotto-tda) (3.7.2)
Requirement already satisfied: plotly>=4.4.1 in /opt/homebrew/lib/python3.11/site-packages (from giotto-tda) (5.17.0)
Requirement already satisfied: ipywidgets>=7.5.1 in /opt/homebrew/lib/python3.11/site-packages (from giotto-tda) (8.1.2)
Collecting python-igraph (from giotto-tda)
  Using cached python_igraph-0.11.4-py3-none-any.whl.metadata (2.8 kB)
Requirement already satisfied: comm>=0.1.3 in /opt/homebrew/lib/python3.11/site-packages (from ipywidgets>=7.5.1->giotto-tda) (0.1.4)
Requirement already satisfied: ipython>=6.1.0 in /opt/homebrew/lib/python3.11/site-packages (from ipywidgets>=7.5.1->giotto-tda) (8.20.0)
Requirement already satisfied: traitlets>=4.3.1 in /opt/homebrew/lib/python3.11/site-packages (from ipywidgets>=7.5.1->giotto-tda) (5.9.0)
Requirement already satisfied: widgetsnbextension~=4.0.10 in /opt/homebrew/lib/python3.11/site-packages (from ipywidgets>=7.5.1->giotto-tda) (4.0.10)
Requirement already satisfied: jupyterlab-widgets~=3.0.10 in /opt/homebrew/lib/python3.11/site-packages (from ipywidgets>=7.5.1->giotto-tda) (3.0.10)
Requirement already satisfied: contourpy>=1.0.1 in /opt/homebrew/lib/python3.11/site-packages (from matplotlib>=3.0.3->giotto-tda) (1.1.0)
Requirement already satisfied: cycler>=0.10 in /opt/homebrew/lib/python3.11/site-packages (from matplotlib>=3.0.3->giotto-tda) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in /opt/homebrew/lib/python3.11/site-packages (from matplotlib>=3.0.3->giotto-tda) (4.42.1)
Requirement already satisfied: kiwisolver>=1.0.1 in /opt/homebrew/lib/python3.11/site-packages (from matplotlib>=3.0.3->giotto-tda) (1.4.5)
Requirement already satisfied: packaging>=20.0 in /opt/homebrew/lib/python3.11/site-packages (from matplotlib>=3.0.3->giotto-tda) (23.2)
Requirement already satisfied: pillow>=6.2.0 in /opt/homebrew/lib/python3.11/site-packages (from matplotlib>=3.0.3->giotto-tda) (10.2.0)
Requirement already satisfied: pyparsing<3.1,>=2.3.1 in /opt/homebrew/lib/python3.11/site-packages (from matplotlib>=3.0.3->giotto-tda) (2.4.7)
Requirement already satisfied: python-dateutil>=2.7 in /opt/homebrew/lib/python3.11/site-packages (from matplotlib>=3.0.3->giotto-tda) (2.8.2)
Requirement already satisfied: tenacity>=6.2.0 in /opt/homebrew/lib/python3.11/site-packages (from plotly>=4.4.1->giotto-tda) (8.2.3)
Requirement already satisfied: threadpoolctl>=2.0.0 in /opt/homebrew/lib/python3.11/site-packages (from scikit-learn>=0.22.0->giotto-tda) (3.2.0)
Collecting igraph==0.11.4 (from python-igraph->giotto-tda)
  Using cached igraph-0.11.4-cp39-abi3-macosx_11_0_arm64.whl.metadata (3.9 kB)
Collecting texttable>=1.6.2 (from igraph==0.11.4->python-igraph->giotto-tda)
  Using cached texttable-1.7.0-py2.py3-none-any.whl.metadata (9.8 kB)
Requirement already satisfied: decorator in /opt/homebrew/lib/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (5.1.1)
Requirement already satisfied: jedi>=0.16 in /opt/homebrew/lib/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (0.19.0)
Requirement already satisfied: matplotlib-inline in /opt/homebrew/lib/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (0.1.6)
Requirement already satisfied: prompt-toolkit<3.1.0,>=3.0.41 in /opt/homebrew/lib/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (3.0.43)
Requirement already satisfied: pygments>=2.4.0 in /opt/homebrew/lib/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (2.16.1)
Requirement already satisfied: stack-data in /opt/homebrew/lib/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (0.6.2)
Requirement already satisfied: pexpect>4.3 in /opt/homebrew/lib/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (4.8.0)
Requirement already satisfied: six>=1.5 in /opt/homebrew/lib/python3.11/site-packages (from python-dateutil>=2.7->matplotlib>=3.0.3->giotto-tda) (1.16.0)
Requirement already satisfied: parso<0.9.0,>=0.8.3 in /opt/homebrew/lib/python3.11/site-packages (from jedi>=0.16->ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (0.8.3)
Requirement already satisfied: ptyprocess>=0.5 in /opt/homebrew/lib/python3.11/site-packages (from pexpect>4.3->ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (0.7.0)
Requirement already satisfied: wcwidth in /opt/homebrew/lib/python3.11/site-packages (from prompt-toolkit<3.1.0,>=3.0.41->ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (0.2.6)
Requirement already satisfied: executing>=1.2.0 in /opt/homebrew/lib/python3.11/site-packages (from stack-data->ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (1.2.0)
Requirement already satisfied: asttokens>=2.1.0 in /opt/homebrew/lib/python3.11/site-packages (from stack-data->ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (2.2.1)
Requirement already satisfied: pure-eval in /opt/homebrew/lib/python3.11/site-packages (from stack-data->ipython>=6.1.0->ipywidgets>=7.5.1->giotto-tda) (0.2.2)
Using cached python_igraph-0.11.4-py3-none-any.whl (9.1 kB)
Using cached igraph-0.11.4-cp39-abi3-macosx_11_0_arm64.whl (1.7 MB)
Using cached texttable-1.7.0-py2.py3-none-any.whl (10 kB)
Building wheels for collected packages: giotto-tda
  Building wheel for giotto-tda (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [240 lines of output]
      /opt/homebrew/lib/python3.11/site-packages/setuptools/dist.py:498: SetuptoolsDeprecationWarning: Invalid dash-separated options
      !!

              ********************************************************************************
              Usage of dash-separated 'description-file' will not be supported in future
              versions. Please use the underscore name 'description_file' instead.

              This deprecation is overdue, please update your project and remove deprecated
              calls to avoid build errors in the future.

              See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
              ********************************************************************************

      !!
        opt = self.warn_dash_deprecation(opt, section)
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-13-arm64-cpython-311
      creating build/lib.macosx-13-arm64-cpython-311/gtda
      copying gtda/_version.py -> build/lib.macosx-13-arm64-cpython-311/gtda
      copying gtda/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda
      copying gtda/pipeline.py -> build/lib.macosx-13-arm64-cpython-311/gtda
      copying gtda/base.py -> build/lib.macosx-13-arm64-cpython-311/gtda
      creating build/lib.macosx-13-arm64-cpython-311/gtda/diagrams
      copying gtda/diagrams/_metrics.py -> build/lib.macosx-13-arm64-cpython-311/gtda/diagrams
      copying gtda/diagrams/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/diagrams
      copying gtda/diagrams/features.py -> build/lib.macosx-13-arm64-cpython-311/gtda/diagrams
      copying gtda/diagrams/distance.py -> build/lib.macosx-13-arm64-cpython-311/gtda/diagrams
      copying gtda/diagrams/preprocessing.py -> build/lib.macosx-13-arm64-cpython-311/gtda/diagrams
      copying gtda/diagrams/_utils.py -> build/lib.macosx-13-arm64-cpython-311/gtda/diagrams
      creating build/lib.macosx-13-arm64-cpython-311/gtda/externals
      copying gtda/externals/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals
      creating build/lib.macosx-13-arm64-cpython-311/gtda/tests
      copying gtda/tests/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/tests
      copying gtda/tests/test_pipeline.py -> build/lib.macosx-13-arm64-cpython-311/gtda/tests
      creating build/lib.macosx-13-arm64-cpython-311/gtda/mapper
      copying gtda/mapper/cover.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper
      copying gtda/mapper/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper
      copying gtda/mapper/visualization.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper
      copying gtda/mapper/nerve.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper
      copying gtda/mapper/cluster.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper
      copying gtda/mapper/pipeline.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper
      copying gtda/mapper/filter.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper
      creating build/lib.macosx-13-arm64-cpython-311/gtda/utils
      copying gtda/utils/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/utils
      copying gtda/utils/_docs.py -> build/lib.macosx-13-arm64-cpython-311/gtda/utils
      copying gtda/utils/testing.py -> build/lib.macosx-13-arm64-cpython-311/gtda/utils
      copying gtda/utils/validation.py -> build/lib.macosx-13-arm64-cpython-311/gtda/utils
      creating build/lib.macosx-13-arm64-cpython-311/gtda/time_series
      copying gtda/time_series/embedding.py -> build/lib.macosx-13-arm64-cpython-311/gtda/time_series
      copying gtda/time_series/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/time_series
      copying gtda/time_series/features.py -> build/lib.macosx-13-arm64-cpython-311/gtda/time_series
      copying gtda/time_series/preprocessing.py -> build/lib.macosx-13-arm64-cpython-311/gtda/time_series
      copying gtda/time_series/multivariate.py -> build/lib.macosx-13-arm64-cpython-311/gtda/time_series
      copying gtda/time_series/target.py -> build/lib.macosx-13-arm64-cpython-311/gtda/time_series
      creating build/lib.macosx-13-arm64-cpython-311/gtda/meta_transformers
      copying gtda/meta_transformers/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/meta_transformers
      copying gtda/meta_transformers/features.py -> build/lib.macosx-13-arm64-cpython-311/gtda/meta_transformers
      creating build/lib.macosx-13-arm64-cpython-311/gtda/graphs
      copying gtda/graphs/geodesic_distance.py -> build/lib.macosx-13-arm64-cpython-311/gtda/graphs
      copying gtda/graphs/transition.py -> build/lib.macosx-13-arm64-cpython-311/gtda/graphs
      copying gtda/graphs/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/graphs
      copying gtda/graphs/kneighbors.py -> build/lib.macosx-13-arm64-cpython-311/gtda/graphs
      creating build/lib.macosx-13-arm64-cpython-311/gtda/homology
      copying gtda/homology/point_clouds.py -> build/lib.macosx-13-arm64-cpython-311/gtda/homology
      copying gtda/homology/consistent.py -> build/lib.macosx-13-arm64-cpython-311/gtda/homology
      copying gtda/homology/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/homology
      copying gtda/homology/_utils.py -> build/lib.macosx-13-arm64-cpython-311/gtda/homology
      creating build/lib.macosx-13-arm64-cpython-311/gtda/diagrams/tests
      copying gtda/diagrams/tests/test_features.py -> build/lib.macosx-13-arm64-cpython-311/gtda/diagrams/tests
      copying gtda/diagrams/tests/test_distance.py -> build/lib.macosx-13-arm64-cpython-311/gtda/diagrams/tests
      copying gtda/diagrams/tests/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/diagrams/tests
      copying gtda/diagrams/tests/test_preprocessing.py -> build/lib.macosx-13-arm64-cpython-311/gtda/diagrams/tests
      creating build/lib.macosx-13-arm64-cpython-311/gtda/externals/python
      copying gtda/externals/python/cubical_complex_interface.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python
      copying gtda/externals/python/simplex_tree_interface.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python
      copying gtda/externals/python/strong_witness_complex_interface.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python
      copying gtda/externals/python/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python
      copying gtda/externals/python/periodic_cubical_complex_interface.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python
      copying gtda/externals/python/witness_complex_interface.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python
      copying gtda/externals/python/rips_complex_interface.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python
      copying gtda/externals/python/ripser_interface.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python
      copying gtda/externals/python/cech_complex_interface.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python
      creating build/lib.macosx-13-arm64-cpython-311/gtda/externals/python/tests
      copying gtda/externals/python/tests/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python/tests
      copying gtda/externals/python/tests/test_witness.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python/tests
      copying gtda/externals/python/tests/test_rips_complex.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python/tests
      copying gtda/externals/python/tests/test_cech_complex.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python/tests
      copying gtda/externals/python/tests/test_periodic_cubical_complex.py -> build/lib.macosx-13-arm64-cpython-311/gtda/externals/python/tests
      creating build/lib.macosx-13-arm64-cpython-311/gtda/mapper/utils
      copying gtda/mapper/utils/_logging.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper/utils
      copying gtda/mapper/utils/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper/utils
      copying gtda/mapper/utils/_cover.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper/utils
      copying gtda/mapper/utils/visualization.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper/utils
      copying gtda/mapper/utils/_cluster.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper/utils
      copying gtda/mapper/utils/pipeline.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper/utils
      copying gtda/mapper/utils/_list_feature_union.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper/utils
      copying gtda/mapper/utils/decorators.py -> build/lib.macosx-13-arm64-cpython-311/gtda/mapper/utils
      creating build/lib.macosx-13-arm64-cpython-311/gtda/utils/tests
      copying gtda/utils/tests/test_validation.py -> build/lib.macosx-13-arm64-cpython-311/gtda/utils/tests
      copying gtda/utils/tests/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/utils/tests
      creating build/lib.macosx-13-arm64-cpython-311/gtda/time_series/tests
      copying gtda/time_series/tests/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/time_series/tests
      copying gtda/time_series/tests/test_multivariate.py -> build/lib.macosx-13-arm64-cpython-311/gtda/time_series/tests
      copying gtda/time_series/tests/test_embedding.py -> build/lib.macosx-13-arm64-cpython-311/gtda/time_series/tests
      copying gtda/time_series/tests/test_preprocessing.py -> build/lib.macosx-13-arm64-cpython-311/gtda/time_series/tests
      creating build/lib.macosx-13-arm64-cpython-311/gtda/meta_transformers/tests
      copying gtda/meta_transformers/tests/test_features.py -> build/lib.macosx-13-arm64-cpython-311/gtda/meta_transformers/tests
      copying gtda/meta_transformers/tests/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/meta_transformers/tests
      creating build/lib.macosx-13-arm64-cpython-311/gtda/graphs/tests
      copying gtda/graphs/tests/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/graphs/tests
      copying gtda/graphs/tests/test_transition.py -> build/lib.macosx-13-arm64-cpython-311/gtda/graphs/tests
      copying gtda/graphs/tests/test_geodesic_distance.py -> build/lib.macosx-13-arm64-cpython-311/gtda/graphs/tests
      copying gtda/graphs/tests/test_kneighbors.py -> build/lib.macosx-13-arm64-cpython-311/gtda/graphs/tests
      creating build/lib.macosx-13-arm64-cpython-311/gtda/homology/tests
      copying gtda/homology/tests/__init__.py -> build/lib.macosx-13-arm64-cpython-311/gtda/homology/tests
      copying gtda/homology/tests/test_consistent.py -> build/lib.macosx-13-arm64-cpython-311/gtda/homology/tests
      copying gtda/homology/tests/test_point_clouds.py -> build/lib.macosx-13-arm64-cpython-311/gtda/homology/tests
      running build_ext
      Cloning into '/private/var/folders/jw/48lxzfqs2v1bgvc5vqznd_gw0000gp/T/pip-install-jpyovsn4/giotto-tda_fd82ebed974843d7a7570f2169f00a20/gtda/externals/pybind11'...
      -- The CXX compiler identification is AppleClang 14.0.3.14030022
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- pybind11 v2.12.0 dev1
      -- CMake 3.27.4
      -- Found Python: /opt/homebrew/Frameworks/Python.framework/Versions/3.12/bin/python3.12 (found suitable version "3.12.2", minimum required is "3.6") found components: Interpreter Development.Module Development.Embed
      -- Python 3.12.2
      -- Performing Test HAS_FLTO_THIN
      -- Performing Test HAS_FLTO_THIN - Failed
      -- Performing Test HAS_FLTO
      -- Performing Test HAS_FLTO - Success
      -- pybind11::lto enabled
      -- pybind11::thin_lto enabled
      -- Configuring done (1.0s)
      -- Generating done (0.0s)
      -- Build files have been written to: /private/var/folders/jw/48lxzfqs2v1bgvc5vqznd_gw0000gp/T/pip-install-jpyovsn4/giotto-tda_fd82ebed974843d7a7570f2169f00a20/gtda/externals/pybind11/build
      Install the project...
      -- Install configuration: ""
      -- Up-to-date: /usr/local/include/pybind11
      CMake Error at cmake_install.cmake:41 (file):
        file INSTALL cannot set permissions on "/usr/local/include/pybind11":
        Operation not permitted.


      make: *** [install] Error 1
      Install the project...
      -- Install configuration: ""
      -- Up-to-date: /usr/local/include/pybind11
      -- Installing: /usr/local/include/pybind11/attr.h
      -- Installing: /usr/local/include/pybind11/embed.h
      -- Installing: /usr/local/include/pybind11/numpy.h
      -- Installing: /usr/local/include/pybind11/pybind11.h
      -- Installing: /usr/local/include/pybind11/operators.h
      -- Installing: /usr/local/include/pybind11/iostream.h
      -- Installing: /usr/local/include/pybind11/gil.h
      -- Installing: /usr/local/include/pybind11/chrono.h
      -- Installing: /usr/local/include/pybind11/typing.h
      -- Installing: /usr/local/include/pybind11/stl_bind.h
      -- Installing: /usr/local/include/pybind11/buffer_info.h
      -- Installing: /usr/local/include/pybind11/options.h
      -- Up-to-date: /usr/local/include/pybind11/eigen
      -- Installing: /usr/local/include/pybind11/eigen/matrix.h
      -- Installing: /usr/local/include/pybind11/eigen/common.h
      -- Installing: /usr/local/include/pybind11/eigen/tensor.h
      -- Up-to-date: /usr/local/include/pybind11/stl
      -- Installing: /usr/local/include/pybind11/stl/filesystem.h
      -- Installing: /usr/local/include/pybind11/functional.h
      -- Installing: /usr/local/include/pybind11/stl.h
      -- Up-to-date: /usr/local/include/pybind11/detail
      -- Installing: /usr/local/include/pybind11/detail/type_caster_base.h
      -- Installing: /usr/local/include/pybind11/detail/typeid.h
      -- Installing: /usr/local/include/pybind11/detail/descr.h
      -- Installing: /usr/local/include/pybind11/detail/internals.h
      -- Installing: /usr/local/include/pybind11/detail/common.h
      -- Installing: /usr/local/include/pybind11/detail/class.h
      -- Installing: /usr/local/include/pybind11/detail/init.h
      -- Installing: /usr/local/include/pybind11/common.h
      -- Installing: /usr/local/include/pybind11/eval.h
      -- Installing: /usr/local/include/pybind11/cast.h
      -- Installing: /usr/local/include/pybind11/type_caster_pyobject_ptr.h
      -- Installing: /usr/local/include/pybind11/eigen.h
      -- Installing: /usr/local/include/pybind11/gil_safe_call_once.h
      -- Installing: /usr/local/include/pybind11/pytypes.h
      -- Installing: /usr/local/include/pybind11/complex.h
      -- Installing: /usr/local/share/cmake/pybind11/pybind11Config.cmake
      -- Installing: /usr/local/share/cmake/pybind11/pybind11ConfigVersion.cmake
      -- Installing: /usr/local/share/cmake/pybind11/FindPythonLibsNew.cmake
      -- Installing: /usr/local/share/cmake/pybind11/pybind11Common.cmake
      -- Installing: /usr/local/share/cmake/pybind11/pybind11Tools.cmake
      -- Installing: /usr/local/share/cmake/pybind11/pybind11NewTools.cmake
      -- Installing: /usr/local/share/cmake/pybind11/pybind11Targets.cmake
      -- Installing: /usr/local/share/pkgconfig/pybind11.pc
      fatal: not a git repository (or any of the parent directories): .git
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/jw/48lxzfqs2v1bgvc5vqznd_gw0000gp/T/pip-install-jpyovsn4/giotto-tda_fd82ebed974843d7a7570f2169f00a20/setup.py", line 183, in <module>
          setup(name=DISTNAME,
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/dist.py", line 989, in run_command
          super().run_command(command)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/opt/homebrew/lib/python3.11/site-packages/wheel/bdist_wheel.py", line 364, in run
          self.run_command("build")
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/dist.py", line 989, in run_command
          super().run_command(command)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/command/build.py", line 131, in run
          self.run_command(cmd_name)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/dist.py", line 989, in run_command
          super().run_command(command)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/private/var/folders/jw/48lxzfqs2v1bgvc5vqznd_gw0000gp/T/pip-install-jpyovsn4/giotto-tda_fd82ebed974843d7a7570f2169f00a20/setup.py", line 116, in run
          self.install_dependencies()
        File "/private/var/folders/jw/48lxzfqs2v1bgvc5vqznd_gw0000gp/T/pip-install-jpyovsn4/giotto-tda_fd82ebed974843d7a7570f2169f00a20/setup.py", line 150, in install_dependencies
          subprocess.check_call(['git', 'submodule', 'update',
        File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 413, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['git', 'submodule', 'update', '--init', '--recursive']' returned non-zero exit status 128.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for giotto-tda
  Running setup.py clean for giotto-tda
Failed to build giotto-tda
ERROR: Could not build wheels for giotto-tda, which is required to install pyproject.toml-based projects
```

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
Example: "No error should be thrown."
-->
No error should be thrown.
**Actual behaviour**
<!--
Please paste or specifically describe the actual output or traceback.
Alternatively, add screenshots to help explain your problem.
-->

**Versions**

Please run the following snippet and paste the output below.
```
import platform; print(platform.platform()): 'macOS-13.2.1-arm64-arm-64bit'
import sys; print("Python", sys.version): Python 3.12.2 (main, Feb  6 2024, 20:19:44) [Clang 15.0.0 (clang-1500.1.0.2.5)]
import numpy; print("NumPy", numpy.__version__): 1.26.1
import scipy; print("SciPy", scipy.__version__): '1.11.3'
import joblib; print("Joblib", joblib.__version__): Joblib 1.3.2
import sklearn; print("Scikit-learn", sklearn.__version__): 1.3.1
import gtda; print("Giotto-tda", gtda.__version__): Cannot install

```

**Additional context**
<!--
Add any other context about the problem here.
-->

<!-- Thanks for contributing! -->

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -20,7 +20,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
-        python-version: [3.8, 3.9, '3.10', '3.11']
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12']
         include:
           - os: ubuntu-latest
             path: ~/.cache/pip
@@ -44,7 +44,7 @@ jobs:
       - uses: actions/checkout@v2
 
       - uses: actions/setup-python@v2
-        name: Install Python-${{ matrix.python-version }} 
+        name: Install Python-${{ matrix.python-version }}
         with:
           python-version: ${{ matrix.python-version }}
 
@@ -110,7 +110,7 @@ jobs:
       ######################
       #  Building package  #
       ######################
-      
+
       - name: Build ${{ github.event.repository.name }} on Linux and MacOs
         if: ${{ runner.os != 'Windows' }}
         run: |
@@ -140,7 +140,7 @@ jobs:
       #############
       #  Testing  #
       #############
-      
+
       - name: Test package with pytest on Linux and Windows
         if: ${{ runner.os == 'Linux' || runner.os == 'Windows' }}
         run: |
@@ -182,7 +182,7 @@ jobs:
       ######################
       #  Upload Artifacts  #
       ######################
-      
+
       - name: Upload coverage for Mac
         if: ${{ runner.os == 'macOS' }}
         uses: actions/upload-artifact@v2

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -73,9 +73,9 @@ jobs:
         env:
           # Specify which Python versions to build wheels
           # https://cibuildwheel.readthedocs.io/en/stable/options/#build-skip
-          CIBW_BUILD: "cp38-* cp39-* cp310-* cp311-*"
-          # Skip 32 bit architectures, musllinux, and i686, and macOS x86_64 wheels for CP3.8 -- CP3.11
-          CIBW_SKIP: "*-win32 *-musllinux_x86_64 *_i686 cp38-macosx_x86_64 cp39-macosx_x86_64 cp310-macosx_x86_64 cp311-macosx_x86_64"
+          CIBW_BUILD: "cp38-* cp39-* cp310-* cp311-* cp312-*"
+          # Skip 32 bit architectures, musllinux, and i686, and macOS x86_64 wheels for CP3.8 -- CP3.12
+          CIBW_SKIP: "*-win32 *-musllinux_x86_64 *_i686 cp38-macosx_x86_64 cp39-macosx_x86_64 cp310-macosx_x86_64 cp311-macosx_x86_64 cp312-macosx_x86_64"
           CIBW_BEFORE_BUILD_WINDOWS: python -m pip install --upgrade pip setuptools && sed -i $'s/\r$//' README.rst && python -m pip install delvewheel
           CIBW_BEFORE_BUILD_LINUX: python -m pip install --upgrade pip setuptools
           CIBW_BEFORE_BUILD_MACOS: python -m pip install --upgrade pip setuptools
@@ -88,7 +88,7 @@ jobs:
           CIBW_ENVIRONMENT_PASS_LINUX: BOOST_ROOT
           CIBW_REPAIR_WHEEL_COMMAND_WINDOWS: "delvewheel repair -vv -w {dest_dir} {wheel}"
           CIBW_MANYLINUX_X86_64_IMAGE: manylinux2014
-          # Should generate universal2 wheels for CP3.8 -- CP3.11
+          # Should generate universal2 wheels for CP3.8 -- CP3.12
           CIBW_ARCHS_MACOS: x86_64 universal2
 
       - name: Set-up python 3.10 for upload

--- a/gtda/_version.py
+++ b/gtda/_version.py
@@ -19,4 +19,4 @@ Topological Data Analysis in a machine learning framework."""
 # 'X.Y.dev0' is the canonical version of 'X.Y.dev'
 #
 
-__version__ = "0.6.1"
+__version__ = "0.6.2"

